packages/data-store/src/contact/ContactStore.ts

```diff
@@ -0,0 +1,305 @@
+      where: { id: contactId },
+    })
+
+    if (!result) {
+      return Promise.reject(Error(`No contact found for id: ${contactId}`))
+    }
+
+    return this.contactFrom(result)
+  }
+
+  getContacts = async (args?: IGetContactsArgs): Promise<Array<IContact>> => {
+    const result = await (await this.dbConnection).getRepository(ContactEntity).find({
+      ...(args?.filter && { where: args?.filter }),
+    })
+
+    return result.map((contact: ContactEntity) => this.contactFrom(contact))
+  }
+
+  addContact = async ({ name, alias, uri }: IAddContactArgs): Promise<IContact> => {
+    if (!name || /^\s*$/.test(name)) {
+      return Promise.reject(Error('Blank names are not allowed'))
+    }
+
+    if (!alias || /^\s*$/.test(alias)) {
+      return Promise.reject(Error('Blank aliases are not allowed'))
+    }
+
+    const result = await (await this.dbConnection).getRepository(ContactEntity).findOne({
+      where: [{ name }, { alias }],
+    })
+
+    if (result) {
+      return Promise.reject(Error(`Duplicate names or aliases are not allowed. Name: ${name}, Alias: ${alias}`))
+    }
+
+    const contactEntity = contactEntityFrom({ name, alias, uri })
+    debug('Adding contact', name)
+    const createdResult = await (await this.dbConnection).getRepository(ContactEntity).save(contactEntity)
+
+    return this.contactFrom(createdResult)
+  }
+
+  updateContact = async ({ contact }: IUpdateContactArgs): Promise<IContact> => {
+    const result = await (await this.dbConnection).getRepository(ContactEntity).findOne({
+      where: { id: contact.id },
+    })
+
+    if (!result) {
+      return Promise.reject(Error(`No contact found for id: ${contact.id}`))
+    }
+
+    const updatedContact = {
+      ...contact,
+      identities: result.identities,
+    }
+
+    debug('Updating contact', contact)
+    const updatedResult = await (await this.dbConnection).getRepository(ContactEntity).save(updatedContact, { transaction: true })
+
+    return this.contactFrom(updatedResult)
+  }
+
+  removeContact = async ({ contactId }: IRemoveContactArgs): Promise<void> => {
+    debug('Removing contact', contactId)
+    ;(await this.dbConnection)
+      .getRepository(ContactEntity)
+      .delete({ id: contactId })
+      .catch((error) => Promise.reject(Error(`Unable to remove contact with id: ${contactId}. ${error}`)))
+  }
+
+  getIdentity = async ({ identityId }: IGetIdentityArgs): Promise<IIdentity> => {
+    const result = await (await this.dbConnection).getRepository(IdentityEntity).findOne({
+      where: { id: identityId },
+    })
+
+    if (!result) {
+      return Promise.reject(Error(`No identity found for id: ${identityId}`))
+    }
+
+    return this.identityFrom(result)
+  }
+
+  getIdentities = async ({ contactId }: IGetIdentitiesArgs): Promise<Array<IIdentity>> => {
+    const result = await (await this.dbConnection).getRepository(ContactEntity).findOne({
```

Comment:
Hmzz, i would expect a method where I can get all kinds of identities with a where filter. And an optional contactId, where basically it is restricted to that contactId. Then also returning potentially multiple matches for that contact

Reply:
it works on a filter now, that also accepts a contactId if you like